### PR TITLE
MB-63992 - High selectivity specific optimisations.

### PR DESCRIPTION
**Pull request by metonymic-smokey**
This PR adds minimises adding vector IDs to a bitmap, even in batches, for high selectivity cases.
It instead adds ineligible vector IDs to the bitmap.

### CODE REVIEW

**Comment by abhinavdangeti** [comment]
@metonymic-smokey would you get this PR ready to go?

File: faiss_vector_posting.go [hunk 1]
```diff
@@ -310,7 +310,7 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 	// Params needed for the closures
 	var vecIndex *faiss.IndexImpl
 	var vecDocIDMap map[int64]uint32
-	var docVecIDMap map[uint32][]int64
+	var docVecIDMap map[int64][]uint32
```
Comment (abhinavdangeti) [inline faiss_vector_posting.go:313]: Why the change in type for the key in this map?

Comment (metonymic-smokey) [reply faiss_vector_posting.go:313]: good point, it can remain uint32.

File: faiss_vector_posting.go [hunk 2]
```diff
@@ -432,72 +432,88 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
...
-									if except != nil && !except.Contains(docID) {
-										ineligibleVectorIDs = append(ineligibleVectorIDs,
-											int64(vecID))
-									} else {
-										ineligibleVectorIDs = append(ineligibleVectorIDs,
-											int64(vecID))
-									}
+							if _, exists := eligibleDocIDsMap[uint64(docID)]; !exists {
+								for _, vecID := range vecIDs {
+									ineligibleVecIDsBitmap.Add(uint32(vecID))
```
Comment (abhinavdangeti) [inline faiss_vector_posting.go:458]: Didn't you want to replace this kind of a thing with `AddMany` instead?

Comment (abhinavdangeti) [reply faiss_vector_posting.go:458]: But that'd mean making another []uint32 array, so ok I guess.
